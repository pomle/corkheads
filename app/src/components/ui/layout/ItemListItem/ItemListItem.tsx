import React from "react";
import { makeStyles } from "@material-ui/styles";

const useStyles = makeStyles({
  itemListItem: {
    alignItems: "center",
    display: "flex",
    minHeight: 50,
    opacity: (props: ItemListItemProps) => (props.disabled ? 0.2 : 1),
    padding: "10px 20px",
    pointerEvents: (props: ItemListItemProps) =>
      props.disabled ? "none" : "all",
  },
  content: {
    width: "100%",
  },
});

interface ItemListItemProps {
  onClick?: () => void;
  disabled?: boolean;
}

const ItemListItem: React.FC<ItemListItemProps> = ({
  onClick,
  disabled,
  children,
}) => {
  const classes = useStyles({ disabled });

  const content = (
    <div className={classes.itemListItem}>
      <div className={classes.content}>{children}</div>
    </div>
  );

  if (onClick) {
    return (
      <button type="button" onClick={onClick} className={classes.content}>
        {content}
      </button>
    );
  }

  return content;
};

export default ItemListItem;
