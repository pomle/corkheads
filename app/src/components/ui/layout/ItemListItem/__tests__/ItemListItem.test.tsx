import React from "react";
import renderer from "react-test-renderer";
import ItemListItem from "../ItemListItem";

describe("ItemListItem", () => {
  it("wraps children", () => {
    const component = renderer.create(
      <ItemListItem>I am a child</ItemListItem>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders as button when onClick supplied", () => {
    const component = renderer.create(
      <ItemListItem onClick={() => undefined} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("invokes onClick when clicked", () => {
    const handleClick = jest.fn();
    const component = renderer.create(<ItemListItem onClick={handleClick} />);
    component.root.findByType("button").props.onClick();
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it("can be disabled", () => {
    const component = renderer.create(<ItemListItem disabled />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
