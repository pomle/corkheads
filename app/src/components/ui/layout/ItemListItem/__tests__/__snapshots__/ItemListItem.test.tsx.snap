// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemListItem can be disabled 1`] = `
<div
  className="makeStyles-itemListItem-1 makeStyles-itemListItem-6"
>
  <div
    className="makeStyles-content-2"
  />
</div>
`;

exports[`ItemListItem renders as button when onClick supplied 1`] = `
<button
  className="makeStyles-content-2"
  onClick={[Function]}
  type="button"
>
  <div
    className="makeStyles-itemListItem-1 makeStyles-itemListItem-4"
  >
    <div
      className="makeStyles-content-2"
    />
  </div>
</button>
`;

exports[`ItemListItem wraps children 1`] = `
<div
  className="makeStyles-itemListItem-1 makeStyles-itemListItem-3"
>
  <div
    className="makeStyles-content-2"
  >
    I am a child
  </div>
</div>
`;
